app/src/main/java/com/jerboa/ui/components/common/PictrsImage.kt

```diff
@@ -152,6 +153,24 @@ fun PickImage(
         mutableStateOf<Bitmap?>(null)
     }
 
+    if (image != null) {
+        LaunchedEffect(image) {
+            imageUri = image
+            // Set the bitmap
+            if (Build.VERSION.SDK_INT < 28) {
+                bitmap.value = MediaStore.Images
+                    .Media.getBitmap(ctx.contentResolver, imageUri!!)
+            } else {
+                val source = ImageDecoder
+                    .createSource(ctx.contentResolver, imageUri!!)
+                bitmap.value = ImageDecoder.decodeBitmap(source)
+            }
+
+            Log.d("jerboa", imageUri.toString())
+            onPickedImage(image)
+        }
```

Comment:
Have you tested this? For me it doesn't actually seem to be doing the image upload.

Reply:
Tested it on firefox and from my gallery, note that when sharing from browsers you have to use 'share image' just sharing the link won't do, but it makes sense to detect if the link is an image, i'll add that shortly

Reply:
My bad, this does work correctly. Not sure why it didn't the first time for me.